### ISSUE TITLE
Bound value in a conditional binding must be of Optional type in Xcode 6.3

### ISSUE THREAD

**Issue by datomnurdin** [jpotts18/SwiftValidator#21]
As per subject, I got this error message on this line of code

```
import Foundation

class PhoneNumberValidation: Validation {
    let PHONE_REGEX = "^\\d{3}-\\d{3}-\\d{4}$"
    
    func validate(value: String) -> (Bool, ValidationErrorType) {
        if let phoneTest = NSPredicate(format: "SELF MATCHES %@", PHONE_REGEX) { //error here
            if phoneTest.evaluateWithObject(value) {
                return (true, .NoError)
            }
            return (false, .PhoneNumber)
        }
        return (false, .PhoneNumber)
    }
    
}
```

Please advice. Thank you.

**Comment by jpotts18** [comment]
The library has changed significantly since this time. What line is the error on?

**Comment by datomnurdin** [comment]
I already state it inside the question.

**Comment by jpotts18** [comment]
Validation Objects have been phased out and replaced with Rules in 2.0

https://github.com/jpotts18/SwiftValidator/blob/master/Validator/PhoneNumberRule.swift

**Comment by imothee** [comment]
Hi,

This problem still exists even on the code you linked to. What the error is saying is that you are using the if let when there is not an optional type returned.

The valid condition is to not have the if and just use the straight let.

```swift
class PhoneNumberRule: Rule {
//    let PHONE_REGEX = "^\\d{3}-\\d{3}-\\d{4}$"
    let PHONE_REGEX = "^\\d{10}$"
    
    var message:String {
        return "Enter a valid 10 digit phone number"
    }
    
    func validate(value: String) -> Bool {
        let phoneTest = NSPredicate(format: "SELF MATCHES %@", PHONE_REGEX)
        if phoneTest.evaluateWithObject(value) {
            return true
        }
        return false
    }
    
    func errorMessage() -> String {
        return message
    }
    
}
```

Xcode Version 6.3 (6D570)

**Comment by datomnurdin** [comment]
Same goes to this one

```
import Foundation

public class RegexRule : Rule {
    
    private var REGEX: String = "^(?=.*?[A-Z]).{8,}$"
    
    public init(regex: String){
        self.REGEX = regex
    }
    
    public func validate(value: String) -> Bool {
        if let test = NSPredicate(format: "SELF MATCHES %@", self.REGEX) {
            if test.evaluateWithObject(value) {
                return true
            }
        }
        return false
    }
    
    public func errorMessage() -> String {
        return "Invalid Regular Expression"
    }
}
```

**Comment by schan93** [comment]
Can confirm this error. I am also getting this issue.

**Comment by datomnurdin** [comment]
Please review my pull request here. 

https://github.com/jpotts18/SwiftValidator/pull/22

**Comment by jpotts18** [comment]
Solved with #22 Thanks!
